onnxruntime/core/providers/coreml/model/model.mm

```diff
@@ -542,13 +405,96 @@ Status Predict(const std::unordered_map<std::string, OnnxTensorData>& inputs,
 Status Execution::Predict(const std::unordered_map<std::string, OnnxTensorData>& inputs,
                           const std::unordered_map<std::string, OnnxTensorInfo>& outputs,
                           const GetOutputTensorMutableRawDataFn& get_output_tensor_mutable_raw_data_fn) {
-  ORT_RETURN_IF_NOT(model_loaded, "Execution::Predict requires Execution::LoadModel");
+  ORT_RETURN_IF_NOT(model_ != nil, "Execution::Predict requires Execution::LoadModel");
 
   if (HAS_COREML3_OR_LATER) {
     @autoreleasepool {
-      return [execution_ predict:inputs
-                         outputs:outputs
-           getOutputTensorDataFn:get_output_tensor_mutable_raw_data_fn];
+      Status status = Status::OK();
+      ORT_TRY {
+        if (model_ == nil) {
```

Comment:
nit: there are two `model_ != nil` checks in this function, we probably can do with just one.